---a/net/minecraft/inventory/IInventory.java
+++b/net/minecraft/inventory/IInventory.java
@@ -1,9 +1,13 @@
 package net.minecraft.inventory;
 
+import java.util.List;
+
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
 import net.minecraft.world.IWorldNameable;
 
-public interface IInventory extends IWorldNameable
-{
+import org.bukkit.craftbukkit.entity.CraftHumanEntity;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.inventory.InventoryHolder;
+
+public interface IInventory extends IWorldNameable {
+
     int getSizeInventory();
 
     ItemStack getStackInSlot(int index);
@@ -35,4 +35,20 @@
     int getFieldCount();
 
     void clear();
-}
+
+    // CraftBukkit start
+    ItemStack[] getContents();
+
+    void onOpen(CraftHumanEntity who);
+
+    void onClose(CraftHumanEntity who);
+
+    List<HumanEntity> getViewers();
+
+    InventoryHolder getOwner();
+
+    void setMaxStackSize(int size);
+
+    int MAX_STACK = 64;
+    // CraftBukkit end
+}
